src/zeroband/inference/config.py

```diff
@@ -1,97 +1,411 @@
+            default=None,
+            description="Maximum model context length. If None, will use the maximum context length from model config.",
+        ),
+    ]
+
+    quantization: Annotated[
+        Literal["awq", "gguf", "gptq", "bitsandbytes", "fp8"] | None,
+        Field(
+            default=None,
+            description="Method used to quantize the weights. If None, will apply the default quantization (if any) from model config.",
+        ),
+    ]
+
+    enforce_eager: Annotated[
+        bool,
+        Field(
+            default=False,
+            description="Whether to enforce eager mode. If False, will use PyTorch eager and cuda graphs in hybrid for maximal performance.",
+        ),
+    ]
+
+    device: Annotated[Literal["auto", "cuda", "cpu"], Field(default="auto", description="Device to use for inference.")]
+
+    enable_thinking: Annotated[
+        bool,
+        Field(default=True, description="Whether to enable thinking. Used by the `format_prompts` function to prepend a thinking prompt."),
+    ]
 
 
 class DifficultyFilteringConfig(BaseConfig):
-    solve_rate_field: str = "solve_rate_qwen_r1_distill_7b"
-    min_solve_rate: float = 0.0
-    max_solve_rate: float = 0.5
+    """Configures filtering of the dataset by difficulty. If None, no filtering is applied."""
+
+    solve_rate_field: Annotated[
+        str, Field(default="solve_rate_qwen_r1_distill_7b", description="Dataset field in the dataset that contains the solve rate.")
+    ]
+    min_solve_rate: Annotated[float, Field(default=0.0, ge=0, le=1, description="Minimum solve rate to include.")]
+    max_solve_rate: Annotated[float, Field(default=0.5, ge=0, le=1, description="Maximum solve rate to include.")]
+
+
+class DataConfig(BaseConfig):
+    """Configures the data to be used for inference."""
+
+    name: Annotated[
+        str,
+        Field(
+            default="PrimeIntellect/INTELLECT-2-RL-Dataset",
+            description="Name of the HF dataset to use.",
+        ),
+    ]
+
+    split: Annotated[str, Field(default="train", description="Split of the dataset to use.")]
+
+    max_prompt_len: Annotated[
+        int | None,
+        Field(
+            default=None,
+            description="If set, filters out all samples with more than this number of input tokens. If None, no filtering is applied.",
+        ),
+    ]
+
+    difficulty_filtering: Annotated[DifficultyFilteringConfig | None, Field(default=None)]
+
+    def __str__(self) -> str:
+        max_prompt_len_str = "disabled" if self.max_prompt_len is None else self.max_prompt_len
+        difficult_filter_str = "disabled" if self.difficulty_filtering is None else self.difficulty_filtering
+        return f"name={self.name} split={self.split} max_prompt_len={max_prompt_len_str} difficulty_filtering={difficult_filter_str}"
+
 
+class RLConfig(BaseConfig):
+    """Configures inference when used in conjunction with a RL trainer. If None, inference will run in a non-RL setting."""
```

Comment:
```suggestion
    """Configures inference when used in conjunction with a RL trainer. """
```

RLConfig should not know about the behavior of the top class